docs/developer/tutorials/snmp/profiles.md

```diff
@@ -0,0 +1,156 @@
+
+- 10-40 metrics is a good amount already.
+- Explore base profiles to see which ones could be applicable to the device.
+- Explore manufacturer-specific MIB files looking for metrics such as:
+    - General health: status gauges...
+    - Network traffic: bytes in/ou, errors in/out, ...
+    - CPU and memory usage.
+    - Temperature: temperature sensors, thermal condition, ...
+    - Power supply.
+    - Storage.
+    - Field-replaceable units ([FRU](https://en.wikipedia.org/wiki/Field-replaceable_unit)).
+    - ...
+
+## Implementation
+
+It might be tempting to gather as many metrics as possible, and only then start building the profile and writing tests.
+
+But we recommend you **start small**. This will allow you to quickly gain confidence on the various components of the SNMP development workflow:
+
+- Editing profile files.
+- Writing tests.
+- Building and using simulation data.
+
+### Add a profile file
+
+Add a `.yaml` file for the profile with the `sysobjectid` and a metric (you'll be able to add more later).
+
+For example:
+
+```yaml
+sysobjectid: 1.3.6.1.4.1.232.9.4.10
+
+metrics:
+  - MIB: CPQHLTH-MIB
+    symbol:
+      OID: 1.3.6.1.4.1.232.6.2.8.1
+      name: cpqHeSysUtilLifeTime
+```
+
+!!! tip
+    `sysobjectid` can also be a wildcard pattern to match a sub-tree of devices, eg `1.3.6.1.131.12.4.*`.
+
+### Add unit tests
+
+Add a unit test in `test_profiles.py` to verify that the metric is successfully collected by the integration when the profile is enabled. (These unit tests are mostly used to prevent regressions and will help with maintenance.)
+
+For example:
+
+```python
+def test_hp_ilo4(aggregator):
+    run_profile_check('hp_ilo4')
+
+    common_tags = common.CHECK_TAGS + ['snmp_profile:hp-ilo4']
+
+    aggregator.assert_metric('snmp.cpqHeSysUtilLifeTime', metric_type=aggregator.MONOTONIC_COUNT, tags=common_tags, count=1)
+    aggregator.assert_all_metrics_covered()
+```
+
+We don't have simulation data yet, so the test should fail. Let's make sure it does:
+
+```console
+$ ddev -k test_hp_ilo4 snmp:py38
+[...]
+======================================== FAILURES ========================================
+_____________________________________ test_hp_ilo42 ______________________________________
+tests/test_profiles.py:1464: in test_hp_ilo4
+    aggregator.assert_metric('snmp.cpqHeSysUtilLifeTime', metric_type=aggregator.GAUGE, tags=common.CHECK_TAGS, count=1)
+../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:253: in assert_metric
+    self._assert(condition, msg=msg, expected_stub=expected_metric, submitted_elements=self._metrics)
+../datadog_checks_base/datadog_checks/base/stubs/aggregator.py:295: in _assert
+    assert condition, new_msg
+E   AssertionError: Needed exactly 1 candidates for 'snmp.cpqHeSysUtilLifeTime', got 0
+[...]
+```
+
+Good. Now, onto adding simulation data.
+
+### Add simulation data
+
+Add a `.snmprec` file named after the `community_string`:
```

Comment:
Should we encourage then to name `.snmprec` to the profile name.

Reply:
Might be a possible follow-up refactoring, but for now I clarified this a bit (it's not so much the community string than the value passed to the `run_check_profile()` helper).